### PR TITLE
Add examples to loss function docstrings

### LINKED ISSUE #166295: Feedback about torch.nn.HingeEmbeddingLoss, HuberLoss, SmoothL1Loss, SoftMarginLoss, MultiLabelSoftMarginLoss Need adding an example
### 📚 The doc issue

There are following issues on these pages: 
https://docs.pytorch.org/docs/main/generated/torch.nn.HingeEmbeddingLoss.html
https://docs.pytorch.org/docs/main/generated/torch.nn.HuberLoss.html
https://docs.pytorch.org/docs/main/generated/torch.nn.SmoothL1Loss.html
https://docs.pytorch.org/docs/main/generated/torch.nn.SoftMarginLoss.html
https://docs.pytorch.org/docs/main/generated/torch.nn.MultiLabelSoftMarginLoss.html

Need adding examples.

I will raise a PR for the same soon.

### Suggest a potential alternative/fix

_No response_

### PR DESCRIPTION
**Pull request by vishalgoyal316**
Added usage examples to the following loss function classes:
- HingeEmbeddingLoss
- SmoothL1Loss
- HuberLoss
- SoftMarginLoss
- MultiLabelSoftMarginLoss

Each example demonstrates basic usage with input tensor creation, loss computation, and backward pass.

Fixes #166295

### CODE REVIEW

**Review by mikaylagawarecki — APPROVED** [review]
Thanks

**Review by mlazos — APPROVED** [review]
